architecture/high-availability.md

```diff
@@ -0,0 +1,33 @@
+# High Availability
+
+Kubecost 2.2 introduces a new flag `haMode` for the frontend service.
+
+This is the first step in enabling high availability for Kubecost. Future versions will expand the HA configuration to include the backend services.
+
+> This flag changes the service name that the ingress needs to target.
+
+With the Frontend HA Mode, the frontend will now be a dedicated deployment with two replicas. This allows for the ability to do rolling upgrades of the frontend pods during upgrades and configuration changes.
+
+The primary benefit to this design is that the Kubecost UI is always available and diagnostics are available even if the backend services are not healthy.

```

Comment:
available used twice, any creative wording you can think of @bstuder99 ?

Reply:
Slightly reworded this